### PR TITLE
Disable vSphere-specific storage tests

### PR DESCRIPTION
**Pull request by bertinatto**
We rely on the standard storage test suite to test vSphere storage.

CC @openshift/storage

### CODE REVIEW

File: test/doc.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
```
Comment (bertinatto) [inline test/doc.go:1]: make update did this change.

File: test/extended/util/annotate/rules.go [hunk 2]
```diff
@@ -68,7 +68,10 @@ var (
 			`\[sig-network-edge\]\[Feature:Idling\] Unidling should work with TCP \(while idling\)`,
 		},
 		// tests that may work, but we don't support them
-		"[Disabled:Unsupported]": {},
+		"[Disabled:Unsupported]": {
+			// https://bugzilla.redhat.com/show_bug.cgi?id=2019115: skip failing vSphere-specific storage tests
```
Comment (jsafrane) [inline test/extended/util/annotate/rules.go:72]: Maybe add a note that tests with `In-tree Volumes [Driver: vsphere]` will still tests the volume plugin enough.

Comment (bertinatto) [reply test/extended/util/annotate/rules.go]: Added a note

**Comment by jsafrane** [comment]
For approvers, fixed vsphere cloud detection in https://github.com/openshift/origin/pull/26551 enabled a huge amount of tests in https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/vsphere that can be run only by VMware folks (they require specific setup). We want to skip them. As pointed out in a comment, vSphere in-tree volume plugin will still be tested by https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/testsuites

**Review by soltysh — APPROVED** [review]
/approve

File: test/extended/util/annotate/rules.go [hunk 3]
```diff
@@ -68,7 +68,12 @@ var (
 			`\[sig-network-edge\]\[Feature:Idling\] Unidling should work with TCP \(while idling\)`,
 		},
 		// tests that may work, but we don't support them
-		"[Disabled:Unsupported]": {},
+		"[Disabled:Unsupported]": {
+			// Skip vSphere-specific storage tests. The standard in-tree storage tests for vSphere
+			// (prefixed with `In-tree Volumes [Driver: vsphere]`) are enough for testing this plugin.
+			// https://bugzilla.redhat.com/show_bug.cgi?id=2019115
+			`\[sig-storage\].*\[Feature:vsphere\]`,
```
Comment (deads2k) [inline test/extended/util/annotate/rules.go:75]: This doesn't seem to match to me.  I see a lot of `[sig-storage] In-tree Volumes [Driver: vsphere] [Testpattern:****` failures.

Comment (deads2k) [reply test/extended/util/annotate/rules.go:75]: Actually, it's more targeted than that.  Some of those do work, but some fail consistently.

Comment (bertinatto) [reply test/extended/util/annotate/rules.go:75]: @deads2k the regex above matches `Feature:vsphere`, not `Driver:vsphere`. We don't want to skip the latter.

Comment (rvanderp3) [reply test/extended/util/annotate/rules.go:75]: @bertinatto there are currently 13 `Driver:vsphere` tests that are broken due to:
~~~
Failed to bootstrap vSphere with error: env variable 'VSPHERE_CONF_FILE' is not set
~~~

It looks like we are intending to skip these [tests](https://github.com/openshift/origin/pull/26565/files#diff-998be43366fe821c61ca242aa34949870c9c6df2572cc060000e4cd990a72bebR80) but those test still appear to be getting run.

Comment (bertinatto) [reply test/extended/util/annotate/rules.go:75]: @rvanderp3 basically we are skipping two set of vSphere tests:
1. All `Feature:vsphere` tests.
2. [Only 4](https://github.com/openshift/origin/pull/26565/files#diff-998be43366fe821c61ca242aa34949870c9c6df2572cc060000e4cd990a72bebR77-R80) `Driver:vsphere` tests that require topology, which we don't support in our CI.

All the remaining `Driver:vsphere` tests should not be skipped and I'm trying to fix the failing ones [here](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22993/rehearse-22993-pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere/1456694637673582592).

Here's a rehearsal job that passed: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22993/rehearse-22993-pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere/1456694637673582592

Comment (rvanderp3) [reply test/extended/util/annotate/rules.go:75]: > @rvanderp3 basically we are skipping two set of vSphere tests:
> 
> 1. All `Feature:vsphere` tests.
> 2. [Only 4](https://github.com/openshift/origin/pull/26565/files#diff-998be43366fe821c61ca242aa34949870c9c6df2572cc060000e4cd990a72bebR77-R80) `Driver:vsphere` tests that require topology, which we don't support in our CI.
> 
> All the remaining `Driver:vsphere` tests should not be skipped and I'm trying to fix the failing ones [here](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22993/rehearse-22993-pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere/1456694637673582592).
> 
> Here's a rehearsal job for that PR: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22993/rehearse-22993-pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere/1456694637673582592

@bertinatto ok, I started a [PR](https://github.com/openshift/origin/pull/26575) to skip those which may not be necessary at this point.  Sorry for the noise!  I'll close that out.

**Pull request merged**